yaksh/models.py

```diff
@@ -2062,6 +2063,73 @@ def get_question_answers(self):
                 }]
         return q_a
 
+    def get_diff_of_consecutive_submissions_for_all_code_question(self):
+        """
+            Return diff of consecutive submissions for all questions
+        """
+        q_a = {} # key : question, value : all submitted answers for this question
+        for answer in self.answers.all():
+            question = answer.question
+            if question in q_a:
+                q_a[question].append(answer)
+            else:
+                q_a[question] = [answer]
+
+        q_diff = {} # key : question, value : list of diff between consecutive submission for this question
+        first_submissions = []
+        for question in q_a:
+            if question.type=='code':
+                answers = q_a[question]
+                all_submissions_for_a_question = [] #all answers for a question as list of lines
+                for ans in answers:
+                    split_ans_line_by_line = [ele+'\n' for ele in ans.answer.split("\n")]
+                    all_submissions_for_a_question.append(split_ans_line_by_line)
+
+                if len(all_submissions_for_a_question)>=1:
+                    first_submissions.append(all_submissions_for_a_question[0])
+                if len(all_submissions_for_a_question)>1:#if there are more than one submissions for this question
+                    zipped = zip(all_submissions_for_a_question, all_submissions_for_a_question[1:])
+                    temp = []
+                    for idx, consecutive_submissions in enumerate(zipped):
+                        diff_of_consecutive_submissions =  difflib.ndiff(consecutive_submissions[0], consecutive_submissions[1])
+                        #remove unwanted ? produced by difflib.ndiff()
+                        temp.append([ele for ele in diff_of_consecutive_submissions if not ele.startswith("?")])
+                    q_diff[question] = zip(temp, all_submissions_for_a_question[1:])
+
+        
+        idx1 =1
+        idx2 =1
+        q_all_diff = {}
+        for idx3, question in enumerate(q_diff):
+            questionwise_diff = '</br><font color="red" size="4"><b>'+question.summary+'</b></font></br>'+''.join(first_submissions[idx3])+\
+            '<hr style="border-top: 3px solid #ccc; background: transparent;">' 
+            idx1 += 1
+            for diff_original in q_diff[question]:
+                diff = diff_original[0]
+                original = diff_original[1]
+                idx2 += 1
```

Comment:
What is idx2/idx1? Provide a proper variable name.

Reply:
these two variables (idx1 : interating variable for outer loop and idx2 : iterating variable for inner loop) are used for generating unique identifiers which will be used in nav-tab